src/Surfnet/StepupGateway/GatewayBundle/Sso2fa/CookieService.php

```diff
@@ -183,71 +181,16 @@ public function shouldSkip2faAuthentication(
             return false;
         }
 
-        /** @var SecondFactor $secondFactor */
-        foreach ($secondFactorCollection as $secondFactor) {
-            $loa = $secondFactor->getLoaLevel($this->secondFactorTypeService);
-            if ($loa >= $requiredLoa) {
-                $this->logger->notice('Verified the current 2FA authentication can be given with the SSO on 2FA cookie');
-                $responseContext->saveSelectedSecondFactor($secondFactor);
-                return true;
-            }
-        }
-        return false;
+        $this->logger->notice('Verified the current 2FA authentication can be given with the SSO on 2FA cookie');
+        $responseContext->saveSelectedSecondFactor($secondFactor);
+        return true;
```

Comment:
This improvement removed a bug that would potentially upgrade the LoA written to the cookie. 

The cookie saved to the state handler would be the first LoA that meets the required LoA (see the removed loop). And that is not the token used per-se.

The intention of this code block was to verify if the LoA is met, but that check is already implemented in the isCookieValid method.